core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -335,4 +335,26 @@ package object config {
         "spark.")
       .booleanConf
       .createWithDefault(false)
+
+  private[spark] val SHUFFLE_FILE_BUFFER_SIZE =
+    ConfigBuilder("spark.shuffle.file.buffer")
+      .doc("Used to set the buffer size of outputstream in shuffle writer.")
+      .bytesConf(ByteUnit.KiB)
+      .checkValues(Set(4, 8, 16, 32, 64, 128, 256, 512, 1024))
```

Comment:
do we have this requirement before? I think the only requirement is, the value should be `>= 0` and `<= Int.MaxValue / 1024`.